test/bitcoin.core.spec.ts

```diff
@@ -214,6 +214,16 @@ describe('Bitcoin-core', () => {
             (hash.reverse() as Buffer).toString('hex'),
             expectedHash,
           );
+
+          assert.doesNotThrow(() =>
+            transaction.hashForWitnessV0(
+              inIndex,
+              script,
+              0,
+              // convert to UInt32
+              hashType < 0 ? 0xffffffff + hashType : hashType,
```

Comment:
`0x100000000 + -1 === 0xffffffff` since -1 should be all f in hex.
`0x100000000 + -2147483648 === 0x80000000` since -2147483648 should be only sign bit.

```suggestion
              hashType < 0 ? 0x100000000 + hashType : hashType,
```

Reply:
ah yes of course

Reply:
I would have preferred `>>> 0` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Unsigned_right_shift) but it doesn't play nice with `tslint` apparently